### DISCUSSION TITLE
Using an arg to set a nested arg

### DISCUSSION THREAD

**Discussion by pdevito3** [storybookjs/storybook#13361] [Help]
So I have a story in storybook like so:

```jsx
export const One = Template.bind();
One.args = {
  children: [
    <MyComponent {...Icon.args} headerText={"Header"} />,
    <MyComponent {...Standard.args}  />
  ]
};
```

Say the `Standard.args` I'm getting look like this:
```jsx
Standard.args = {
  headerText: "Default Header",
  detailText: "Default details"
};
```

I'd like to be able to set the `detailText` value using a distinct arg, but don't know of a way to do it. Something like the below, but obviously with working syntax:
```jsx
export const One = Template.bind();
One.args = {
  detailText: "new detail text",
  children: [
    <MyComponent {...Icon.args} headerText={"Header"} />,
    <MyComponent {...Standard.args} detailText={detailText} />
  ]
};
```

Is this possible? [Here](https://stackoverflow.com/questions/65112027/using-a-storybook-arg-to-set-a-nested-arg)'s the SO if anyone wants points.

**Comment by shilman** [comment]
Args currently need to be JSON serializable so you'll need to do something like this https://storybook.js.org/docs/react/essentials/controls#fully-custom-args
